src/mono/netcore/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.Mono.cs

```diff
@@ -1,16 +1,39 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
+using System.Runtime.CompilerServices;
+
 
 namespace System.Globalization
 {
     internal partial class GlobalizationMode
     {
-        internal static bool UseNls => true;
-
+        internal static bool UseNls {
+            get {
+                return !Invariant && Interop.Globalization.LoadICU() == 0;
+            }
+        }
         private static bool GetGlobalizationInvariantMode()
         {
-            return GetInvariantSwitchValue();
+            bool invariantEnabled = GetInvariantSwitchValue();
+            if (invariantEnabled)
+                return true;
+
+            LoadICU();
```

Comment:
You don't need to call `LoadICU` here as you're calling `Interop.Globalization.LoadICU()` in the `UseNls` getter.